src/service_args.py

```diff
@@ -78,25 +80,36 @@ def __encode_for_emoji(self, text):
         text = text.encode('latin_1').decode("raw_unicode_escape").encode('utf-16', 'surrogatepass').decode('utf-16')
         return text
 
-    def __replace_chain_name(self, value: str, position: int):
-        chain_key_index = [i for i, n in enumerate(self.service_args_list_customized) if n == '--chain'][position]
-        self.service_args_list_customized[chain_key_index + 1] = value
+    def __set_chain_name(self, value: str, position: int):
+        try:
+            chain_key_index = [i for i, n in enumerate(self.service_args_list_customized) if n == '--chain'][position]
+            self.service_args_list_customized[chain_key_index + 1] = value
+        except IndexError:
+            if position == 0:
+                self.__add_firstchain_args(['--chain', value])
+            elif position == 1:
+                self.__add_secondchain_args(['--chain', value])
```

Comment:
Not clear enough to me how this replaces the original `--chain` in the list of arguments. But I'm probably too tired in my head to read more code today :sweat_smile:

I can get back and check this out again tomorrow.

Reply:
So this enables the use case of setting a spec url in config without setting `--chain`. Is that correct?